### ISSUE TITLE
Custom derives and procedural attribute macros don't show errors

### ISSUE THREAD

**Issue by jonas-schievink** [rust-analyzer/rust-analyzer#9132]
When we expand custom `#[derive]`s and attribute macros, we seem to ignore their errors.

Interestingly, if you do this:

```rust
serde::Deserialize!(struct S {
    #[serde(asd)]
    fld: u8,
});
```

...then the error shows up only if the invocation is inside a function body, not when it's in item position. (of course, the fact that you can call a custom derive like that is itself a bug)

**Comment by jonas-schievink** [comment]
Fixed by https://github.com/rust-analyzer/rust-analyzer/pull/10339
